### DISCUSSION TITLE
Latest version of Sketchware Pro no longer compiles

### DISCUSSION THREAD

**Discussion by TheLabScientist** [Sketchware-Pro/Sketchware-Pro#755] [Q&A]
The latest version of Sketchware Pro and Original Sketchware no longer compiles any project, both old projects and newly created projects. On original Sketchware, the compilation error is just '[]' but on Sketchware Pro, it shows an error shown in the attached screenshot.

![20230209_131616](https://user-images.githubusercontent.com/110190470/217892193-28725539-c7da-4674-9246-b5ea21d47849.jpg)

Both Sketchwares were working fine before but now they no longer want to compile project sources. I am using a device running Android 9.0 (API 28). Is there any way to fix this?

I have cleared the data, uninstalled and reinstalled, restarted my device and even deleted the whole .sketchware folder but the problem still occur.

**Comment by j-px** [comment]
Looks like you enabled some App compatibility changes in Developer Settings for Sketchware Pro, or changed the app's target SDK version to be higher than 28. Either turn all compatibility changes off or disable Developer Options entirely.

**Reply by TheLabScientist** [reply]
I've turned developer options off but the problem is still here. There are no app compatibility settings on my device nor did i change the target SDK version of anything

**Reply by j-px** [reply]
what phone did you use, and what *exact* Sketchware Pro version did you use (e.g. v6.4.0-rc01)?

**Reply by TheLabScientist** [reply]
I use a OneLernDB009 tablet. More info in attached screenshot

![20230209_200114](https://user-images.githubusercontent.com/110190470/217966532-67d89138-90f9-41fb-9091-1b2960d46916.jpg)

**Reply by TheLabScientist** [reply]
Forgot to say i use Sketchware pro v6.4.0-rc05-minApi26

**Reply by j-px** [reply]
Taking into account that even vanilla Sketchware fails to compile code, I'm guessing that your device manufacturer screwed up.

You could try using another app that executes binaries stored in its private app data directory, Termux. If it doesn't work too, then it's an issue not on our side, but your device OS' instead.

**Reply by TheLabScientist** [reply]
Yea termux says: **exec("/data/data/com.termux/files/usr/bin/login"): Permission denied**

It was working fine before and now it says this. Anyway, thanks for your responses. I don't think there is any fix to this.
